ui/src/app/utils/access-utils.tsx

```diff
@@ -360,6 +361,47 @@ export const useIsUserDisabled = () => {
   return disabled;
 };
 
+export const useNeedsToAcceptTOS = () => {
```

Comment:
Cool, I don't think that I have ever seen useEffect used in a custom hook before.